numpy/core/tests/test_multiarray.py

```diff
@@ -6515,6 +6518,31 @@ def test_out_of_order_fields(self):
         with assert_raises(ValueError):
             memoryview(arr)
 
+    def test_max_dims(self):
+        a = np.empty((1,) * 32)
+        self._check_roundtrip(a)
+
+    def test_error_too_many_dims(self):
+        # construct a memoryview with 33 dimensions
+        ct = ctypes.c_uint8
+        for i in range(33):
+            ct = 1 * ct
```

Comment:
Tests fail due to a reference cycle involving `ct`, which is somehow detected in a completely different test. Printing `gc.garbage` in the failing test gives:
```
# ndim = 18
A = numpy.core.tests.test_multiarray.c_ubyte_Array_1_Array_1_Array_1_Array_1_Array_1_Array_1_Array_1_Array_1_Array_1_Array_1_Array_1_Array_1_Array_1_Array_1_Array_1_Array_1_Array_1_Array_1
B = numpy.core.tests.test_multiarray.c_ubyte_Array_1_Array_1_Array_1_Array_1_Array_1_Array_1_Array_1_Array_1_Array_1_Array_1_Array_1_Array_1_Array_1_Array_1_Array_1_Array_1_Array_1

[<class 'A'>,
 (<class 'A'>,
  <class '_ctypes.Array'>,
  <class '_ctypes._CData'>,
  <class 'object'>),
 {'__dict__': <attribute '__dict__' of 'A' objects>,
  '__doc__': None,
  '__module__': 'numpy.core.tests.test_multiarray',
  '__weakref__': <attribute '__weakref__' of 'A' objects>,
  '_length_': 1,
  '_type_': <class 'B'>},
 <attribute '__weakref__' of 'A' objects>,
 <attribute '__dict__' of 'A' objects>]
```

This happens even if we remove the `np.array` call below, so the problem must be in either CPython, or parallelization in pytest